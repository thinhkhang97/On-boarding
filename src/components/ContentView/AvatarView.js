import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import Avatar from 'sharedComponents/Avatar';

type Props = {
  source: string
};
const DEFAULT_SIZE = 254;
class AvatarView extends React.Component<Props> {
  render() {
    const { source } = this.props;
    return (
      <View
        style={{
          width: DEFAULT_SIZE,
          height: DEFAULT_SIZE,
          borderRadius: DEFAULT_SIZE / 2,
          borderColor: '#f5f8fc',
          borderWidth: 54,
          overflow: 'hidden',
          padding: 15,
          backgroundColor: 'white',
          marginBottom: 48
        }}
      >
        <Avatar source={source} style={{ flex: 1, width: '100%' }} />
      </View>
    );
  }
}

const styles = StyleSheet.create({});

export default AvatarView;
