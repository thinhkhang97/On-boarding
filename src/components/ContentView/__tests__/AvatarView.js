import * as React from 'react';
import renderer from 'react-test-renderer';
import AvatarView from '../AvatarView';

test('Render avatar view correctly', ()=>{
    const testAvatarView = renderer.create(<AvatarView/>).toJSON();
    expect(testAvatarView).toMatchSnapshot();
})