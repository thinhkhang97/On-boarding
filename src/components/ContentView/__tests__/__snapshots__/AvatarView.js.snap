// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render avatar view correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#f5f8fc",
      "borderRadius": 127,
      "borderWidth": 54,
      "height": 254,
      "marginBottom": 48,
      "overflow": "hidden",
      "padding": 15,
      "width": 254,
    }
  }
>
  <Image
    style={
      Array [
        Object {
          "borderRadius": 58,
          "height": 116,
          "width": 116,
        },
        Object {
          "flex": 1,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
